models/classes/runner/QtiRunnerService.php

```diff
@@ -589,6 +588,30 @@ public function storeItemResponse(RunnerServiceContext $context, $itemRef, $resp
             } else {
                 \common_Logger::e('Invalid json payload');
             }
+            
+            if (!\taoQtiTest_helpers_TestRunnerUtils::doesAllowSkipping($session)) {
+
+                $similar = 0;
+
+                /** @var ResponseVariable $responseVariable */
+                foreach ($responses as $responseVariable) {
+                    $defaultValue = $responseVariable->getDefaultValue();
+                    $value = $responseVariable->getValue();
+
+                    if (!$defaultValue) {
+                        $class = get_class($value);
+                        $defaultValue = new $class($responseVariable->getBaseType(), array());
```

Comment:
It doesn't work for QtiScalar response type such as QtiInteger for a slider interaction for example. The QtiScalar have only one parameter in the construct which is the value. maybe you should talk to @bugalot to have more insight about this part